### PR TITLE
Fix task cancellation authz on fulfilling cluster

### PR DESCRIPTION
**Pull request by albertzaharovits**
This fixes task cancellation actions (i.e. `internal:admin/tasks/cancel_child`
and `internal:admin/tasks/ban`) not being authorized by the fulfilling cluster.
This can result in orphaned tasks on the fulfilling cluster.

### CODE REVIEW

**Comment by albertzaharovits** [comment]
@n1v0lg The root cause here is that `internal:` actions should be authorized by the local system internal user in  https://github.com/elastic/elasticsearch/blob/6955427699e656f6517e7e1b76a21f2cc8686c06/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L315-L317 , because they are otherwise rejected in https://github.com/elastic/elasticsearch/blob/be502cc319a7032e516818fc42a17d2b82830923/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L535-L539 
But `internal:` actions from a remote cluster are not perceived as originating from the **local** system user (it is the remote system user), in which case they appear as `internal:` actions from a non-system user, which is rejected.

**Review by n1v0lg — APPROVED** [review]
LGTM 👍 We also discussed on Slack:

1. We should backport this to 8.14.1
2. This "breaks" in a mixed cluster setting (e.g., in an 8.13.x and 8.15.0 cluster). However, if the failure mode is simply a new exception (instead of the current authz failed one) it's not worth addressing this, esp. since RCS 2.0 is in beta before 8.13
3. An assertion/log for when a "remote" system user fails authz on an internal action would to nice (either in this PR or a follow-up) to make this easier to detect in the future.

Happy to re-review if point 2 above ends up requiring some additional changes, but as it stands this looks ready to me. Thanks for tracking this down!

**Comment by albertzaharovits** [comment]
@n1v0lg

> This "breaks" in a mixed cluster setting (e.g., in an 8.13.x and 8.15.0 cluster). However, if the failure mode is simply a new exception (instead of the current authz failed one) it's not worth addressing this, esp. since RCS 2.0 is in beta before 8.13

As discussed, I did a manual check for this.
On the **querying** cluster, if the cancellation action is either missing or is unauthorized, the log messages are almost identical: https://github.com/elastic/elasticsearch/blob/17c6230b9f314c6abeba88162981cfd458abbd09/server/src/main/java/org/elasticsearch/tasks/TaskCancellationService.java#L200 . On the **fulfilling** cluster, if the action is un-authorized we see a log entry https://github.com/elastic/elasticsearch/blob/17c6230b9f314c6abeba88162981cfd458abbd09/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L536 but if the action is missing there's no corresponding log entry https://github.com/elastic/elasticsearch/blob/17c6230b9f314c6abeba88162981cfd458abbd09/server/src/main/java/org/elasticsearch/transport/InboundAggregator.java#L47

That means that, given the the fix as it currently stands, it won't make it a worse experience when communicating with clusters that don't have the fix (e.g. 8.14.0). Just the log message will be slightly different.

**Comment by albertzaharovits** [comment]
@n1v0lg 
> An assertion/log for when a "remote" system user fails authz on an internal action would to nice (either in this PR or a follow-up) to make this easier to detect in the future.

I pushed https://github.com/elastic/elasticsearch/pull/109357/commits/1f700ab0f9cc5513eff6efee349c928ee6e3cb12 such that the log message on the **fulfilling** cluster includes the authentication. From that, a keen eye can spot if it's a remote system user or not. For example, this is a sample log error:
```
denying access for [Authentication[effectiveSubject=Subject{version=8676000, user=User[username=test_user,roles=[],fullName=null,email=null,metadata={}], realm={Realm[_es_cross_cluster_access._es_cross_cluster_access] on Node[fulfilling-cluster-0]}, type=CROSS_CLUSTER_ACCESS, metadata={_security_api_key_creator_realm_name=default_file, _security_api_key_limited_by_role_descriptors=org.elasticsearch.common.bytes.BytesArray@1323, _security_api_key_id=0wYw6Y8BRt8c6GrAors2, _security_api_key_type=cross_cluster, _security_cross_cluster_access_authentication=Authentication[effectiveSubject=Subject{version=8676000, user=User[username=_system,roles=[],fullName=null,email=null,metadata={}], realm={Realm[__attach.__attach] on Node[query-cluster-0]}, type=USER, metadata={}},type=INTERNAL], _security_api_key_creator_realm_type=file, _security_api_key_name=cross_cluster_access_key, _security_api_key_role_descriptors=org.elasticsearch.common.bytes.BytesArray@8def201, _security_cross_cluster_access_role_descriptors=[]}},type=API_KEY]] as action [internal:admin/tasks/ban] is not an index or cluster action
```
It's verbose AF, but `_security_cross_cluster_access_authentication=Authentication[effectiveSubject=Subject{version=8676000, user=User[username=_system` tells one that this is a remote system user.

I think this should aid debugging and covers the case of a remote system user getting rejected. WDYT?

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java [hunk 1]
```diff
@@ -138,6 +145,169 @@ public class RemoteClusterSecurityRestIT extends AbstractRemoteClusterSecurityTe
...
+            Map<String, Object> submitAsyncSearchResponseMap = XContentHelper.convertToMap(
+                JsonXContent.jsonXContent,
+                EntityUtils.toString(submitAsyncSearchResponse.getEntity()),
+                false
+            );
+            assertThat(submitAsyncSearchResponseMap.get("is_running"), equalTo(true));
+            String asyncSearchId = (String) submitAsyncSearchResponseMap.get("id");
+            assertThat(asyncSearchId, notNullValue());
+            // wait for the tasks to show up on the querying cluster
+            assertTrue(waitUntil(() -> {
```
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java]: Nice, thanks for the pointer! Pushed https://github.com/elastic/elasticsearch/pull/109357/commits/bb508f58b434b3210acfb5deabefeefa6965d03f

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java [hunk 2]
```diff
@@ -138,6 +145,169 @@ public class RemoteClusterSecurityRestIT extends AbstractRemoteClusterSecurityTe
...
+            // submit a stalling remote async search
+            submitAsyncSearchRequest.setJsonEntity("""
+                {
+                  "query": {
+                    "error_query": {
+                      "indices": [
+                        {
+                          "name": "*:*",
+                          "error_type": "exception",
+                          "stall_time_seconds": 30
```
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java]: Agreed, pushed https://github.com/elastic/elasticsearch/pull/109357/commits/03807fb7ab47522b2ba5da461b7d209da72420ab

**Comment by n1v0lg** [comment]
> I think this should aid debugging and covers the case of a remote system user getting rejected. WDYT?

@albertzaharovits sounds good!

**Pull request merged**